common/persistence/tests/cassandra_test.go

```diff
@@ -1215,3 +1235,265 @@ func deleteMessages(
 
 	return err
 }
+
+func testCassandraNexusIncomingServiceStore(t *testing.T, cluster *cassandra.TestCluster, tableVersion *atomic.Int64) {
+	store := newNexusIncomingServiceStore(cluster.GetSession())
+	t.Run("ConcurrentCreate", func(t *testing.T) {
+		testCassandraNexusIncomingServiceStoreConcurrentCreate(t, store, tableVersion)
+	})
+	t.Run("ConcurrentUpdate", func(t *testing.T) {
+		testCassandraNexusIncomingServiceStoreConcurrentUpdate(t, store, tableVersion)
+	})
+	t.Run("ConcurrentCreateAndUpdate", func(t *testing.T) {
+		testCassandraNexusIncomingServiceStoreConcurrentCreateAndUpdate(t, store, tableVersion)
+	})
+	t.Run("ConcurrentUpdateAndDelete", func(t *testing.T) {
+		testCassandraNexusIncomingServiceStoreConcurrentUpdateAndDelete(t, store, tableVersion)
+	})
+}
+
+func testCassandraNexusIncomingServiceStoreConcurrentCreate(t *testing.T, store persistence.NexusServiceStore, tableVersion *atomic.Int64) {
+	ctx := context.Background()
```

Comment:
Can we put a timeout on the test context?

Reply:
I would rather not right now since we don't have it on any of the other tests AFAICT and they execute fairly quickly

Reply:
If there's an issue with a test, the run may hang.
It's good practice IMHO.